SMBLibrary/Client/SMB2Client.cs

```diff
@@ -21,9 +21,10 @@ public class SMB2Client : ISMBClient
         public static readonly int NetBiosOverTCPPort = 139;
         public static readonly int DirectTCPPort = 445;
 
-        public static readonly uint ClientMaxTransactSize = 1048576;
-        public static readonly uint ClientMaxReadSize = 1048576;
-        public static readonly uint ClientMaxWriteSize = 1048576;
+        private static readonly uint RawDataBufferSize = 1048576 - 256 - SessionPacket.HeaderLength;

```

Comment:
ArrayPool buffer size is of power of 2. 
Here we reduce the buffer size by MaxTransactSize  and HeaderLength. 
Buffer has now 1Mib instead of 2Mib per request.